.github/workflows/python-package.yml

```
@@ -0,0 +1,49 @@
+name: Python package
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - 2.7
+          - 3.5
+          - 3.6
+          - 3.9
+          - pypy2
+          - pypy3
+
+    env:
+      TOX_PARALLEL_NO_SPINNER: 1
+      TOXENV: python
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: pip install tox
+      - name: "Initialize tox envs: ${{ matrix.env.TOXENV }}"
+        run: >-
+          python -m
+          tox
+          --parallel auto
+          --parallel-live
+          --notest
+          --skip-missing-interpreters false
+      - name: Test with tox
+        run: >-
+          python -m
+          tox
+          --parallel auto
+          --parallel-live
+          --
+          -vvvvv
```

Let's also add a job for linting. It uses matrix as inputs because it's more maintainable even though it generates just a single job. It makes it easier to add extra tox envs via the matrix spec vs finding what to update in the middle of the steps:

```suggestion
          -vvvvv

lint:
  runs-on: ubuntu-latest
  strategy:
    matrix:
      python-version:
      - 3.9
      toxenv:
      - style

  env:
    TOX_PARALLEL_NO_SPINNER: 1
    TOXENV: ${{ matrix.env }}

  steps:
  - uses: actions/checkout@v2
  - name: Set up Python ${{ matrix.python-version }}
    uses: actions/setup-python@v2
    with:
      python-version: ${{ matrix.python-version }}
  - name: Install dependencies
    run: pip install tox
  - name: "Initialize tox envs: ${{ env.TOXENV }}"
    run: >-
      python -m
      tox
      --parallel auto
      --parallel-live
      --notest
      --skip-missing-interpreters false
  - name: Test with tox
    run: >-
      python -m
      tox
      --parallel auto
      --parallel-live
      --
      -vvvvv
```